ui/appui-react/src/appui-react/preview/new-toolbars/useOverflow.tsx

```diff
@@ -0,0 +1,124 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Toolbar
+ */
+
+import * as React from "react";
+import type { SizeProps } from "@itwin/core-react";
+import { useRefs, useResizeObserver } from "@itwin/core-react";
+
+type Orientation = "horizontal" | "vertical";
+
+/** @internal */
+export const useOverflow = <
+  TContainer extends HTMLElement,
+  TComponent extends HTMLElement
+>(
+  allItems: ReadonlyArray<string>,
+  orientation: Orientation,
+  getItemSize: (item: string) => SizeProps,
+  getOverflowSize: () => SizeProps
+) => {
+  const [visible, setVisible] = React.useState(allItems.length);
+  const [calculate, setCalculate] = React.useState(false);
+  const componentRef = React.useRef<TComponent>();
```

Comment:
Looks like this ref is unused. Only used in `useRefs` call.

Reply:
Good catch, removed.